mupq.py

```diff
@@ -186,33 +228,43 @@ def get_implementations(self, all=False):
 
     @abc.abstractmethod
     def run_test(self, implementation):
+        self.log.info("Runnning %s - %s", implementation, self.test_type)
         implementation.build_binary(f'{self.test_type}',
                                     self.platform_settings.binary_type)
         binary = implementation.get_binary_path(f'{self.test_type}',
                                                 self.platform_settings.binary_type)
-        return self.interface.run(binary)
+        return self.interface.run(binary, self.iterations)
 
     def test_all(self, args=[]):
+        implementations = []
         exclude = "--exclude" in args
         for implementation in self.get_implementations():
             if exclude and implementation.scheme in args:
                 continue
             if not exclude and len(args) > 0 and implementation.scheme not in args:
                 continue
-            if self.run_test(implementation) == -1:
-                return -1
+            implementations.append(implementation)
+        with tqdm.tqdm(total=len(implementations), desc=self.test_type) as pb:
+            for implementation in implementations:
+                pb.set_postfix_str(f"{implementation}")
+                if self.run_test(implementation) == -1:
+                    pb.write(f"{implementation} FAIL")
+                    return -1
+                pb.write(f"{implementation} DONE")
```

Comment:
I think it is better to output 'SUCCESSFUL' rather than 'DONE'.

Reply:
Done